### ISSUE TITLE
The SAM/BAM file declares more reference sequences (572504) than RSEM knows (88647)!

### ISSUE THREAD

**Issue by LHXqwq** [deweylab/RSEM#143]
First, i use STAR to align genome, then i run this command: 
rsem-calculate-expression --paired-end --no-bam-output \
--alignments -p 5 \
-q 03align_out/4w_1/4w_1_Aligned.toTranscriptome.out.bam \
gm_rsem/gm_rsem \
04rsem_out/4w-1_rsem
Finally , it shows "The SAM/BAM file declares more reference sequences (572504) than RSEM knows (88647)!"
How can i solve this problem? Thanks, best wishes!

**Comment by cndewey** [comment]
This error usually occurs when the STAR reference was built with a different set of sequences or gene annotations than that used for RSEM.  Either make sure you are using the same GTF and FASTA files for both STAR and RSEM references or use rsem-prepare-reference with the --star option to build the RSEM and STAR references simultaneously.

**Comment by estelabruxel** [comment]
I had the same problem. I used the same fasta and GTF files. However, to construct reference sequences, RSEM did not identify 67 transcripts. So, when we run calculate expression an error warning: “The SAM/BAM file declares more reference sequences (227954) than RSEM knows (227887)."

**Comment by cndewey** [comment]
Perhaps there are some formatting issues with the GTF files that are preventing RSEM from identifying some transcript annotations.  @estelabruxel - would you be able to share your GTF file or at least the GTF lines for the 67 transcripts that are not identified?

**Comment by estelabruxel** [comment]
Of course, It is Homo_sapiens.GRCh38.100.gtf from Ensembl: ftp://ftp.ensembl.org/pub/release-100/gtf/homo_sapiens
"Warning: Cannot extract transcript ENST00000614535's sequence since the chromosome it locates, GL000220.1, is absent!
Warning: Cannot extract transcript ENST00000616638's sequence since the chromosome it locates, KI270728.1, is absent!
Warning: Cannot extract transcript ENST00000611062's sequence since the chromosome it locates, GL000216.2, is absent!
Warning: Cannot extract transcript ENST00000622674's sequence since the chromosome it locates, KI270727.1, is absent!
Warning: Cannot extract transcript ENST00000400754's sequence since the chromosome it locates, GL000194.1, is absent!
Warning: Cannot extract transcript ENST00000611446's sequence since the chromosome it locates, GL000220.1, is absent!
Warning: Cannot extract transcript ENST00000618003's sequence since the chromosome it locates, KI270726.1, is absent!
Warning: Cannot extract transcript ENST00000612882's sequence since the chromosome it locates, KI270727.1, is absent!
Warning: Cannot extract transcript ENST00000619109's sequence since the chromosome it locates, KI270713.1, is absent!
Warning: Cannot extract transcript ENST00000614351's sequence since the chromosome it locates, KI270721.1, is absent!
Warning: Cannot extract transcript ENST00000618998's sequence since the chromosome it locates, KI270733.1, is absent!
Warning: Cannot extract transcript ENST00000621028's sequence since the chromosome it locates, KI270728.1, is absent!
Warning: Cannot extract transcript ENST00000618083's sequence since the chromosome it locates, KI270713.1, is absent!
Warning: Cannot extract transcript ENST00000617983's sequence since the chromosome it locates, KI270734.1, is absent!
Warning: Cannot extract transcript ENST00000619317's sequence since the chromosome it locates, GL000220.1, is absent!
Warning: Cannot extract transcript ENST00000612465's sequence since the chromosome it locates, GL000195.1, is absent!
Warning: Cannot extract transcript ENST00000615130's sequence since the chromosome it locates, KI270733.1, is absent!
Warning: Cannot extract transcript ENST00000612848's sequence since the chromosome it locates, KI270721.1, is absent!
Warning: Cannot extract transcript ENST00000617089's sequence since the chromosome it locates, KI270721.1, is absent!
Warning: Cannot extract transcript ENST00000616292's sequence since the chromosome it locates, KI270733.1, is absent!
Warning: Cannot extract transcript ENST00000621409's sequence since the chromosome it locates, GL000225.1, is absent!
Warning: Cannot extract transcript ENST00000614604's sequence since the chromosome it locates, KI270727.1, is absent!
Warning: Cannot extract transcript ENST00000620810's sequence since the chromosome it locates, KI270727.1, is absent!
Warning: Cannot extract transcript ENST00000616048's sequence since the chromosome it locates, KI270727.1, is absent!
Warning: Cannot extract transcript ENST00000620032's sequence since the chromosome it locates, GL000205.2, is absent!
Warning: Cannot extract transcript ENST00000615165's sequence since the chromosome it locates, KI270734.1, is absent!
Warning: Cannot extract transcript ENST00000621424's sequence since the chromosome it locates, KI270734.1, is absent!
Warning: Cannot extract transcript ENST00000612925's sequence since the chromosome it locates, KI270750.1, is absent!
Warning: Cannot extract transcript ENST00000613230's sequence since the chromosome it locates, GL000194.1, is absent!
Warning: Cannot extract transcript ENST00000618679's sequence since the chromosome it locates, GL000195.1, is absent!
Warning: Cannot extract transcript ENST00000612315's sequence since the chromosome it locates, KI270713.1, is absent!
Warning: Cannot extract transcript ENST00000616049's sequence since the chromosome it locates, GL000213.1, is absent!
Warning: Cannot extract transcript ENST00000616157's sequence since the chromosome it locates, GL000213.1, is absent!
Warning: Cannot extract transcript ENST00000621137's sequence since the chromosome it locates, KI270727.1, is absent!
Warning: Cannot extract transcript ENST00000612243's sequence since the chromosome it locates, KI270728.1, is absent!
Warning: Cannot extract transcript ENST00000614110's sequence since the chromosome it locates, KI270728.1, is absent!
Warning: Cannot extract transcript ENST00000611746's sequence since the chromosome it locates, KI270728.1, is absent!
Warning: 67 transcripts are failed to extract because their chromosome sequences are absent."

**Comment by cndewey** [comment]
@estelabruxel - In your case, the FASTA files you have provided are missing some of the chromosomal sequences on which these transcripts are annotated.  I would suggest either (1) removing these transcripts from the GTF if you do not care about them or (2) providing the missing chromosomal sequences in the FASTA.

There is potentially an issue here in that in these situations, perhaps STAR is including these transcripts in its reference even though it does not have sequence for them, causing a mismatch with the RSEM reference, which explicitly excludes them.

**Comment by estelabruxel** [comment]
Ok...Thank you so much @cndewey

**Comment by rocanja** [comment]
I think this is conceptually similar to issue: [https://github.com/deweylab/RSEM/issues/49 ](https://github.com/deweylab/RSEM/issues/49).

**Comment by Fayeelw** [comment]
Warning: The SAM/BAM file declares less reference sequences (191563) than RSEM knows (200905)! Please make sure that you aligned your reads against transcript sequences instead of genome.
RSEM can not recognize reference sequence name tr_NC_000001.11_17369_17391_15!
